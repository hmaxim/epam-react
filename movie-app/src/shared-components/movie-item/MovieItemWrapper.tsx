import styled from "styled-components";

const MovieItemWrapper = styled.div`
  width: 323px;
  margin: 40px 20px;

  .movie-info {
    color: white;
    display: flex;
    justify-content: space-between;
    align-items: flex-start;

    .movie-title {
      font-size: 20px;
      font-weight: 600;
    }

    .movie-genre {
    }

    .movie-year {
      padding: 3px 5px;
      border: 1px solid white;
      border-radius: 3px;
    }
  }
`;

MovieItemWrapper.displayName = "MovieItemWrapper";

export default MovieItemWrapper;
