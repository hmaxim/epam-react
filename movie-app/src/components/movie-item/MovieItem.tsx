import React from "react";
import MovieItemWrapper from "../../components/movie-item/MovieItemWrapper";
import posterLogo from "../../assets/images/netflix-streaming-vs-traditional-cable.jpg";
import { IMovie } from "../../interfaces/Movie";

const MovieItem = (props: IMovie) => {
  return (
    <MovieItemWrapper
      onClick={() =>
        console.log(`navigated to details with + ${props.id}`)
      }
    >
      <div className="movie-poster">
        <img src={posterLogo} alt="poster"></img>
      </div>
      <div className="movie-info">
        <div>
          <div className="movie-title">{props.title}</div>
          {props.genres.map((genre: string, index: number) => (
            <span key={index}>{genre}</span>
          ))}
        </div>
        <div className="movie-year">{props.release_date}</div>
      </div>
    </MovieItemWrapper>
  );
};

export default MovieItem;
