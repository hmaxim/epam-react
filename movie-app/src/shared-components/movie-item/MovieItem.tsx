import React, { useCallback } from "react";
import MovieItemWrapper from "./MovieItemWrapper";
import posterLogo from "../../assets/images/netflix-streaming-vs-traditional-cable.jpg";
import { IMovie } from "../../interfaces/Movie";

const MovieItem = (props: IMovie) => {
  const navigate = useCallback(() => {
    console.log(`navigated to details with + ${props.id}`);
  }, []);

  return (
    <MovieItemWrapper onClick={() => navigate}>
      <div className="movie-poster">
        <img src={posterLogo} alt="poster"></img>
      </div>
      <div className="movie-info">
        <div>
          <div className="movie-title">{props.title}</div>
          {props.genres.map((genre: string, index: number) => (
            <span key={index}>{genre}</span>
          ))}
        </div>
        <div className="movie-year">{props.release_date}</div>
      </div>
    </MovieItemWrapper>
  );
};

MovieItem.displayName = "MovieItem";

export default MovieItem;
