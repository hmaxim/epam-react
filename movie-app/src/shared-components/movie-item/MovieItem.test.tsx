import 'jsdom-global/register';
import React, { Component } from 'react';
import { shallow, mount, render } from 'enzyme';
import { IMovie } from '../../interfaces/IMovie';
import MovieItem from './MovieItem';
import 'jest-enzyme';
import 'jest-styled-components';
import MovieItemWrapper from './MovieItemWrapper';

const MOVIE: IMovie = {
  id: 0,
  title: 'Titanic',
  tagline: null,
  vote_average: 4,
  vote_count: 0,
  release_date: '1997',
  poster_path: '../../assets/images/netflix-streaming-vs-traditional-cable.jpg',
  overview:
    "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
  budget: 0,
  revenue: 0,
  runtime: 150,
  genres: ['Action'],
};

describe('MovieItem Component', () => {
  let component: any;

  beforeEach(() => {
    component = shallow(<MovieItem {...MOVIE} />);
  });

  test('should be rendered', () => {
    expect(component.find(MovieItemWrapper)).toMatchSnapshot();
  });

  test('should does click', () => {
    component.find(MovieItemWrapper).simulate('click');
  });
});
