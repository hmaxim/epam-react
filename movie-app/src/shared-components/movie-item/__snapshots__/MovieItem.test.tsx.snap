// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem Component should be rendered 1`] = `
<MovieItemWrapper
  onClick={[Function]}
>
  <div
    className="movie-poster"
  >
    <img
      alt="poster"
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </div>
  <div
    className="movie-info"
  >
    <div>
      <div
        className="movie-title"
      >
        Titanic
      </div>
      <span
        key="0"
      >
        Action
      </span>
    </div>
    <div
      className="movie-year"
    >
      1997
    </div>
  </div>
</MovieItemWrapper>
`;
